src/topology_utils.py

```diff
@@ -223,17 +233,10 @@ def get_contacts(args, urltype, roottype):
                "&active=on&active_value=1&disable=on&disable_value=0"
     with get_auth_session(args) as session:
         url = mangle_url(base_url, args, session)
-        attempts = 3
-        response = None
-        while attempts > 0:
-            try:
-                response = session.get(url)
-            except OSError as exc:
-                attempts -= 1
-                if attempts > 0: print("Incorrect password, please try again")
-        if response is None:
-            print("Too many incorrect password attempts, exiting")
-            exit()
+        try:
+            response = session.get(url)
+        except OSError as exc:
```

Comment:
There are _many_ reasons why an OSError might happen while getting a webpage, not all of which mean incorrect password. You should experiment to find out what specific error you get when you're using an incorrect password, and catch that specific error. I suspect it will be one of the others in this list: https://docs.python-requests.org/en/latest/api/#exceptions

(Wait until topology.opensciencegrid.org is back up before experimenting.)

Reply:
Hey Mat, I experimented with the exceptions but the stacktrace keeps showing that OSError is the exception being thrown and the requests library raises ProtocolError and ConnectionError. I tried catching those exceptions but it doesn't seem to work.